reactor-core/src/test/java21/reactor/core/scheduler/LoomBoundedElasticSchedulerTest.java

```diff
@@ -0,0 +1,267 @@
+			Assertions.assertThat(value1).isEqualTo(queue.poll());
+
+			Object value2 = queue.poll();
+			Assertions.assertThat(value2).isEqualTo(queue.poll());
+			Assertions.assertThat(value2).isEqualTo(queue.poll());
+			worker.dispose();
+		}
+	}
+
+	@Test
+	public void ensuresConcurrentDelayedTasksSchedulingSingleWorker() throws InterruptedException {
+		Queue<Object> queue = Queues.unboundedMultiproducer()
+		                              .get();
+		for (int i = 0; i < 100; i++) {
+			CountDownLatch latch = new CountDownLatch(3);
+
+			Scheduler.Worker worker = scheduler.createWorker();
+
+			RaceTestUtils.race(() -> worker.schedule(() -> {
+				queue.offer("1");
+				queue.offer("1");
+				queue.offer("1");
+				latch.countDown();
+			}, 1, TimeUnit.MILLISECONDS), () -> worker.schedule(() -> {
+				queue.offer("2");
+				queue.offer("2");
+				queue.offer("2");
+				latch.countDown();
+			}), () -> worker.schedule(() -> {
+				queue.offer("3");
+				queue.offer("3");
+				queue.offer("3");
+				latch.countDown();
+			}, 1, TimeUnit.MILLISECONDS));
+
+			Assertions.assertThat(latch.await(5, TimeUnit.SECONDS))
+			          .isTrue();
+
+			Object value1 = queue.poll();
+			Assertions.assertThat(value1).isEqualTo(queue.poll());
+			Assertions.assertThat(value1).isEqualTo(queue.poll());
+
+			Object value2 = queue.poll();
+			Assertions.assertThat(value2).isEqualTo(queue.poll());
+			Assertions.assertThat(value2).isEqualTo(queue.poll());
+
+			Object value3 = queue.poll();
+			Assertions.assertThat(value3).isEqualTo(queue.poll());
+			Assertions.assertThat(value3).isEqualTo(queue.poll());
+			worker.dispose();
+		}
+	}
+
+	@Test
+	public void ensuresConcurrentPeriodicTasksSchedulingSingleWorker() throws InterruptedException {
+		Queue<Object> queue = Queues.unboundedMultiproducer()
+		                            .get();
+		for (int i = 0; i < 100; i++) {
+			CountDownLatch latch = new CountDownLatch(10);
+
+			Scheduler.Worker worker = scheduler.createWorker();
+
+			RaceTestUtils.race(() -> worker.schedulePeriodically(() -> {
+				queue.offer("1");
+				queue.offer("1");
+				queue.offer("1");
+				latch.countDown();
+			}, 1, 0, TimeUnit.MILLISECONDS), () -> worker.schedule(() -> {
+				queue.offer("2");
+				queue.offer("2");
+				queue.offer("2");
+				latch.countDown();
+			}, 1, TimeUnit.MILLISECONDS), () -> worker.schedulePeriodically(() -> {
+				queue.offer("3");
+				queue.offer("3");
+				queue.offer("3");
+				latch.countDown();
+			}, 1, 1, TimeUnit.MILLISECONDS));
+
+			Assertions.assertThat(latch.await(5, TimeUnit.SECONDS))
+			          .isTrue();
+
+			for (int j = 0; j < 10; j++) {
+				Object value = queue.poll();
+				Assertions.assertThat(value)
+				          .isEqualTo(queue.poll());
+				Assertions.assertThat(value)
+				          .isEqualTo(queue.poll());
+			}
+			worker.dispose();
+		}
+	}
+
+	@Test
+	public void ensuresConcurrentWorkerTaskDisposure() throws InterruptedException {
```

Comment:
This one is nice :) Apart from the existing cases, I think it would be great to test:
- schedule adding numbers 1..10k (one by one) to a queue and validate the queue contains numbers in the same order (validates single task is executed and the backing queue holds the tasks in proper order)
- two workers sharing the same BoundedServices instance and disposing one doesn't dispose tasks of the other.